server/src/main/java/org/elasticsearch/index/fieldvisitor/FieldsVisitor.java

```diff
@@ -90,10 +90,6 @@ public Status needsField(FieldInfo fieldInfo) {
     public final void postProcess(Function<String, MappedFieldType> fieldTypeLookup) {
         for (Map.Entry<String, List<Object>> entry : fields().entrySet()) {
             MappedFieldType fieldType = fieldTypeLookup.apply(entry.getKey());
-            if (fieldType == null) {
```

Comment:
I moved this logic to the caller, it is now part of the lookup function that is provided.